chat-client/src/client/mynahUi.ts

```diff
@@ -822,11 +834,18 @@ export const createMynahUi = (
                 const oldMessage = chatItems.find(ci => ci.messageId === updatedMessage.messageId)
                 if (!oldMessage) return
 
+                // Check if this message should be muted
+                const shouldMute =
+                    // Check for status indicating stopped/rejected
+                    updatedMessage.header?.status?.text === 'Stopped' ||
+                    updatedMessage.header?.status?.text === 'Rejected'
+
                 const chatItem: ChatItem = {
                     type: oldMessage.type,
                     ...prepareChatItemFromMessage(updatedMessage, getTabPairProgrammingMode(mynahUi, tabId)),
+                    // Apply muted if the message should be muted
+                    ...(shouldMute ? { muted: true } : {}),
```

Comment:
Same here, if this can be moved to `prepareChatItemFromMessage` then we can simplify this